test/test_Quaternion.jl

```diff
@@ -156,22 +156,78 @@ Base.:(/)(a::MyReal, b::Real) = a.val / b
 # this used to throw an error
 @test qrotation([1, 0, 0], MyReal(1.5)) == qrotation([1, 0, 0], 1.5)
 
-for _ in 1:100
-    let # test specialfunctions
-        c = Complex(randn(2)...)
-        q, q2 = sample(Quaternion{Float64}, 4)
-        unary_funs = [exp, log, sin, cos, sqrt, inv, conj, abs2, norm]
-        # since every quaternion is conjugate to a complex number,
-        # one can establish correctness as follows:
-        for fun in unary_funs
-            @test fun(Quaternion(c)) ≈ Quaternion(fun(c))
+@testset "non-analytic functions" begin
+    q, q2 = randn(Quaternion{Float64}, 2)
+    unary_funs = [conj, abs, abs2, norm, sign]
+    # since every quaternion is conjugate to a complex number,
+    # one can establish correctness as follows:
+    @testset for fun in unary_funs
+        for i in 1:100
+            c = randn(ComplexF64)
+            @test fun(Quaternion(c)) ≈ fun(c)
             @test q2 * fun(q) * inv(q2) ≈ fun(q2 * q * inv(q2))
         end
+    end
+end
 
-        @test exp(log(q)) ≈ q
-        @test exp(zero(q)) ≈ one(q)
+@testset "complex analytic functions" begin
+    # all complex analytic functions can be extended to the quaternions
+    q, q2 = randn(Quaternion{Float64}, 2)
+    unary_funs = [
+        sqrt, inv, exp, exp2, exp10, expm1, log, log2, log10, log1p, cis,
+        sin, cos, tan, asin, acos, atan, sinh, cosh, tanh, asinh, acosh, atanh,
+        csc, sec, cot, acsc, asec, acot, csch, sech, coth, acsch, asech, acoth,
+        sinpi, cospi,
+    ]
+    # since every quaternion is conjugate to a complex number,
+    # one can establish correctness as follows:
+    @testset for fun in unary_funs
+        for i in 1:100
+            c = randn(ComplexF64)
```

Comment:
```suggestion
    unary_funs = [
        sqrt, inv, exp, exp2, exp10, expm1, log, log2, log10, log1p, cis,
        sin, cos, tan, asin, acos, atan, sinh, cosh, tanh, asinh, acosh, atanh,
        csc, sec, cot, acsc, asec, acot, csch, sech, coth, acsch, asech, acoth,
        sinpi, cospi,
    ]
    # since every quaternion is conjugate to a complex number,
    # one can establish correctness as follows:
    @testset "Check $(fun) compatibility with complex number" for fun in unary_funs
        q, q2 = randn(Quaternion{Float64}, 2)
        for _ in 1:100
            c = randn(ComplexF64)
```
I think it would be better to move `q` and `q2` into `@testset` to make the scope narrower and easier to read.